File: backend/cmd/database/models/Crew.go
package models

import "gorm.io/gorm"

type Crew struct {
	gorm.Model
	ArtistID   uint   `json:"artistID"`
	Artist     Artist `json:"artist"`
	MovieID    uint   `json:"movieID"`
	Movie      Movie  `json:"movie"`
	Job        string `gorm:"not null;default:'unknown'" json:"job"`
	Department string `gorm:"not null;default:'unknown'" json:"department"`
}

File: prototype/models/Crew.js
import mongoose from "mongoose";

const Crew = {
  artist: {
    type: [mongoose.Schema.Types.ObjectId],
    ref: "artist",
    required: [true, "artist is required"],
  },

  job: {
    type: String,
    default: "unknown",
    trim: true,
  },

  department: {
    type: String,
    default: "unknown",
    trim: true,
  },
};

export default Crew;
